### ISSUE TITLE
IntraACL+Cirrus or Sphinx Search

### ISSUE THREAD

**Issue by mukulagrawal78** [mediawiki4intranet/IntraACL#80]
I have a quick question. 
If I use Cirrus search engine on a wiki with IntraACL, would I need to patch Cirrus?
I prefer Cirrus over Sphinx, provided it works with IntraACL. Pls help.

Alternatively, I see that you have Sphinx in your distribution of extensions. I don't see any installation instructions. I am assuming your git repository corresponds only to the "extension" piece of the setupI am guessing I will need to separately install the Sphinx Search Engine from linux repositories and then install your extension separately, is that right? Can I follow instruction provided by mediawiki here: https://www.mediawiki.org/wiki/Extension:SphinxSearch.

**Comment by vitalif** [comment]
Yes, every MediaWiki extension usually does direct DB access and should be patched in order to support page-level read permissions. Maybe CirrusSearch already does it, I don't know (it's sufficient to use the standard userCan('read') hook).

Installation instructions for our sphinx extension are here: http://wiki.4intra.net/SphinxSearchEngine

**Comment by vitalif** [comment]
Ooops. It seems instructions are missing

**Comment by vitalif** [comment]
Anyway it's easy. You just install Sphinx, add a config similar to provided sphinx.conf, include SphinxSearchEngine.php and set connection parameters (see inside it). Then you call `php extensions/SphinxSearchEngine/rebuild-sphinx.php --conf LocalSettings.php`, that's all.

**Comment by mukulagrawal78** [comment]
Vitalif: Can you be clear what you mean by the "connection parameters"? I dont see any connection parameters either in the sphinx.conf or in the SphinxSearchEngine.php.

I am getting a whe biunch of warning like below. I am on MW 1.27.

Warning: SphinxQLClient::query(): Couldn't fetch mysqli in /var/lib/wiki/extensions/SphinxSearchEngine/SphinxSearchEngine_class.phon line 954
PHP Warning:  mysqli::close(): Couldn't fetch mysqli in /var/lib/wiki/extensions/SphinxSearchEngine/SphinxSearchEngine_class.php oline 901

**Comment by vitalif** [comment]
Host/port/socket
(Sphinx uses MySQL protocol)
